In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
from scipy.stats import t


## An important quality characteristic used by the manufacturers of ABC asphalt shingles is the amount of moisture the shingles contain when they are packaged. Customers may feel that they have purchased a product lacking in quality if they find moisture and wet shingles inside the packaging.   In some cases, excessive moisture can cause the granules attached to the shingles for texture and colouring purposes to fall off the shingles resulting in appearance problems. To monitor the amount of moisture present, the company conducts moisture tests. A shingle is weighed and then dried. The shingle is then reweighed, and based on the amount of moisture taken out of the product, the pounds of moisture per 100 square feet is calculated. The company claims that the mean moisture content cannot be greater than 0.35 pound per 100 square feet.
#### The file (A & B shingles.csv) includes 36 measurements (in pounds per 100 square feet) for A shingles and 31 for B shingles.

In [2]:
df3 = pd.read_csv('A+&+B+shingles.csv')

In [3]:
df3.describe()

A          B
count  36.000000  31.000000
mean    0.316667   0.273548
std     0.135731   0.137296
min     0.130000   0.100000
25%     0.207500   0.160000
50%     0.290000   0.230000
75%     0.392500   0.400000
max     0.720000   0.580000

In [4]:
## checking for null values
df3.isnull().sum()

A    0
B    5
dtype: int64

## For the A shingles, form the null and alternative hypothesis to test whether the population mean moisture content is less than 0.35 pound per 100 square feet.

In [5]:
tstat, p_value = stats.ttest_1samp(df3['A'],.35)
tstat, p_value

(-1.4735046253382782, 0.14955266289815025)

### Assuming Alpha = 5%

## Hypothesis:
#### Null : Mean moisture content is greater than or equal to 0.35
#### Alternate: Mean moisture content is less that 0.35

## Since P value is 14.95 %, which is greater than alpha for all 3 significance values(99%, 95% and 90%), we fail to reject the null hypothesis.
## Hence, for the A shingles, the population mean moisture content is greater than or equal to 0.35.
## That is, the population mean moisture content is not less than 0.35


## For the B shingles, form the null and alternative hypothesis to test whether the population mean moisture content is less than 0.35 pound per 100 square feet.

In [6]:
tstat_B, p_value_B = stats.ttest_1samp(df3['B'].dropna(),.35)
tstat_B, p_value_B

(-3.1003313069986995, 0.004180954800638363)

### Assuming Alpha = 5%

## Hypothesis:
#### Null : Mean moisture content is greater than or equal to 0.35
#### Alternate: Mean moisture content is less that 0.35

## Since P value is 0.418 %, which is lesser than alpha for all 3 significance values(99%, 95% and 90%), we will reject the null hypothesis.¶
## Hence, for the B shingles, the population mean moisture content is less than 0.35

## Checking if the population means for shingles A and B are equal. Forming the hypothesis and conduct the test of the hypothesis. Listing out the various assumptions we need to check before the test for equality of means is performed

In [7]:
t_stat_equal_means, p_value_equal_means = stats.ttest_ind(df3['A'],df3['B'].dropna())
t_stat_equal_means, p_value_equal_means

(1.289628271966112, 0.2017496571835328)

    
### Assumptions:
1. Data is randomly sampled from the population of interest
2. The data variables follow a normal distribution. Only then one can specify a level of probability (alpha level, level of significance, p) as a criterion for acceptance.
3. Sample size is reasonably large enough to perform the hypothesis tests.
4. The samples are independent of each other
5. Data is continuous
    
### Assuming Alpha = 5%


## Hypothesis:
#### Null : Population means of shingles A and  B are equal
#### Alternate: Population means of shingles A and  B are not equal


## Inference:
### From the T Test of two independent variables, we see that the P value is 20.17%, which is higher than our Alpha value. So, we fail to reject the null hypothesis.

### Hence, the population means for shingles A and B are equal.

## Listing out what assumptions about the population distribution is needed in order to conduct the hypothesis tests above

### Assumptions about population:
#### The data variables of the population follow a normal distribution. Only then one can specify a level of probability (alpha, level of significance, p) as a criterion for acceptance or rejection.
#### Data is continuous

### Assumptions about Samples:
#### Sample data is randomly selcetd from the population
#### Sample size is suffuciently large enough to perform the hypothesis tests.
#### Each sample of population data is independed of each other